File: backend/src/main/java/fr/hoenheimsports/backend/membershipservice/entities/Campaign.java
package fr.hoenheimsports.backend.membershipservice.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

/**
 * Entity representing a membership campaign.
 */
@Entity
@Table(name = "campaigns", schema = "membership_schema")
@Getter
@Setter
@NoArgsConstructor
public class Campaign {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id")
    private UUID id;

    @Column(name = "season_id", nullable = false)
    private UUID seasonId;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private CampaignStatus status;

    @ElementCollection
    @CollectionTable(name = "campaign_categories", schema = "membership_schema", joinColumns = @JoinColumn(name = "campaign_id"))
    private Set<Category> categories = new HashSet<>();
}

File: frontend/projects/shared-domain/src/lib/membership/campaign.ts
import {UUID} from '../uuid';
import {Category} from './category';

/**
 * Enum representing the status of a membership campaign.
 */
export enum CampaignStatus {
  DRAFT = 'DRAFT',
  LAUNCHED = 'LAUNCHED',
  CLOSED = 'CLOSED',
}

/**
 * Represents a membership campaign.
 */
export interface Campaign {
  /** Unique identifier for the campaign. */
  id: UUID;
  /** The identifier of the season associated with this campaign. */
  seasonId: UUID;
  /** The current status of the campaign. */
  status: CampaignStatus;
  /** The list of categories for this campaign. */
  categories: Category[];
}
